src/app/locator/qgsgotolocatorfilter.cpp

```diff
@@ -87,51 +88,114 @@ void QgsGotoLocatorFilter::fetchResults( const QString &string, const QgsLocator
     }
   }
 
-  if ( okX && okY )
+  if ( okX && okY )  // Ok found a valid looking x,y coordinate pair, check against Wgs84crs, MapCanvas-crs AND layer-crs's
   {
-    QVariantMap data;
-    const QgsPointXY point( posX, posY );
-    data.insert( QStringLiteral( "point" ), point );
-
-    const bool withinWgs84 = wgs84Crs.bounds().contains( point );
-    if ( !posIsDms && currentCrs != wgs84Crs )
+    // collect all crs's from all layers (plus MapCanvas crs and wgs84)
+    QList<QString> crsList;  // TODO make this a list
+    crsList.append( wgs84Crs.authid() );
+    if ( currentCrs != wgs84Crs )
+      crsList.append( currentCrs.authid() );
+    for ( const QgsMapLayer *layer : QgsProject::instance()->mapLayers() )
+    {
+      if ( !crsList.contains( layer->crs().authid() ) )
+      {
+        crsList.append( layer->crs().authid() );
+      }
+    }
+    // Now go over all crs's and check if the coordinate COULD be use in that crs
+    for ( const QString &authid : crsList )
     {
+
+      QgsCoordinateReferenceSystem crs = QgsCoordinateReferenceSystem( authid );
+
+      QVariantMap data;
+      const QgsPointXY point( posX, posY );
+      data.insert( QStringLiteral( "point" ), point );
+
       QgsLocatorResult result;
       result.filter = this;
-      result.displayString = tr( "Go to %1 %2 (Map CRS, %3)" ).arg( locale.toString( point.x(), 'g', 10 ), locale.toString( point.y(), 'g', 10 ), currentCrs.userFriendlyIdentifier() );
-      result.userData = data;
-      result.score = 0.9;
-      emit resultFetched( result );
-    }
 
-    if ( withinWgs84 )
-    {
-      if ( currentCrs != wgs84Crs )
+      const bool withinWgs84 = wgs84Crs.bounds().contains( point );
```

Comment:
This should just be checking the current crs from the list:
```suggestion
      const bool coordinateIsWithinCrs = crs.bounds().contains( point );
```

Reply:
ie there shouldn't be a special code path for wgs84

Reply:
Ok, only reason I see there was (and I copied that) is that the output string says:
```"Go to %1° %2° (%3)"```
So there are little 'degree' symbols...

But I can try to remove that codepath and for example add the 'degree' symbol only if geographical... or I just forget about them anyway....